server/db/models/User.js

```diff
@@ -9,34 +9,74 @@ const jwt = require('jsonwebtoken');
 require('dotenv').config();
 
 class User extends Model {
-  authenticate = async function ({ email, password }) {
-    const user = await User.findOne({
-      where: {
-        email,
-      },
+  static authenticate({ email, password }) {
+    let id;
+    return new Promise((res, rej) => {
+      User.findOne({
+        where: {
+          email,
+        },
+      })
+        .then((user) => {
+          if (user) {
+            id = user.id;
+            return bcrypt.compare(password, user.password);
+          }
+          throw new Error('bad credentials');
+        })
+        .then((comparison) => {
+          if (comparison) {
+            const token = jwt.sign({ id }, process.env.ACCESS_TOKEN_SECRET);
+            res(token);
+          }
+          throw new Error('bad credentials');
+        })
+        .catch(() => {
+          const error = new Error('bad credentials');
+          error.status = 401;
+          rej(error);
+        });
+    });
+  }
```

Comment:
A disadvantage of defining instance and model methods in the `class/extends Model` framework is that you cannot use `async` functions to my knowledge. I refactored these to simply return Promises for the sake of getting the auth routes working